libkbfs/prefetcher.go

```diff
@@ -496,6 +586,19 @@ func (p *blockPrefetcher) run(testSyncCh <-chan struct{}) {
 			p.log.Debug("canceling prefetch for block %s", blockID)
 			// Walk up the block tree and delete every parent.
 			p.applyToParentsRecursive(p.cancelPrefetch, blockID, pre)
+		case reqInt := <-p.prefetchRescheduleCh.Out():
+			req := reqInt.(*prefetchRequest)
+			blockID := req.ptr.ID
+			pre, ok := p.prefetches[blockID]
+			if !ok {
+				pre = p.newPrefetch(1, false, req)
+				p.prefetches[blockID] = pre
+			}
+			p.log.Debug("rescheduling top-block prefetch for block %s", blockID)
+			p.applyToParentsRecursive(
+				func(blockID kbfsblock.ID, pp *prefetch) {
+					p.rescheduleTopBlock(blockID, pp, req)
```

Comment:
I'm a bit confused as to why you're passing in `req` here. Can't you just create a new one inside the function? You seem to be reusing the `req` struct and reassigning stuff inside it as you walk up the tree.

Reply:
The problem is all the other fields in the `req`: priority, lifetime, isDeepSync, etc.  I didn't really want to pass all of those around individually, and I didn't know how to figure them out from within the `rescheduleTopBlock` function.  If you know what the right rules should be, I'll gladly change it.

Reply:
I think `pre.req` is more likely to have the correct parameters than `req`, since `pre.req` pertains directly to the `blockID` being traversed in `applyToParentsRecursive`.

Ultimately though this depends on what you want:
1. If you're trying to reschedule the top block prefetch as-is, then use `pre.req`.
2. If you're trying to reschedule the top block based on the parameters in `req`, then use `req`.

I suggested the first approach based on my assumption that we want 1, and that 1 also results in cleaner code.